
syntax = "proto3";

service Worker {
    rpc Infer (InferenceRequest) returns (InferenceResponse){}
    rpc Load (ModelManageRequest) returns (ModelManageResponse){}
    rpc Unload (ModelManageRequest) returns (ModelManageResponse){}
    rpc GetMetrics (MetricsRequest) returns (MetricsResponse){}
}

message InferenceRequest {
    string json_obj = 1;
}

message InferenceResponse {
    enum Status {
        UNAVAILABLE = 0;
        UNLOADED = 1;
        FAILURE = 2;
        SUCCESS = 3;
    }
    string response = 1;
    Status infer_status = 2;
    string error_msg = 3;
    string request_id = 4;
    string json_obj = 5;
}

message ModelManageRequest {
    string model_name=1;
}

message ModelManageResponse {
    enum Status {
        UNAVAILABLE = 0;
        NOT_ENOUGH_SPACE = 1;
        UNLOADED = 2;
        LOADED = 3;
    }
    Status model_status = 1;
    string error_msg = 2;
}

message MetricsRequest {
    // I don't know if there are any details that we'll be submitting yet
    // For instance, specific models to ask about
    // Or the fields that we care about
}

message MetricsResponse {
    string metrics_json = 1;
}

